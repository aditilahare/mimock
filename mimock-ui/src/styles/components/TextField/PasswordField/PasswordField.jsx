import React from 'react';
import { PasswordFieldContainer } from './PasswordField.style';
import PropTypes from 'prop-types';

function PasswordField({
	dataTestid,
	placeholder,
	autoComplete,
	value,
	onChangeHandler,
	onBlurHandler,
	onFocusHandler,
}) {
	return (
		<PasswordFieldContainer
			data-testid={dataTestid}
			placeholder={placeholder}
			autoComplete={autoComplete}
			value={value}
			onChange={onChangeHandler}
			onBlur={onBlurHandler}
			onFocus={onFocusHandler}
		/>
	);
}

PasswordField.propTypes = {
	dataTestid: PropTypes.string.isRequired,
	value: PropTypes.string.isRequired,
	onChangeHandler: PropTypes.func,
	onBlurHandler: PropTypes.func,
	onFocusHandler: PropTypes.func,
	placeholder: PropTypes.string,
	autoComplete: PropTypes.bool,
};

export default PasswordField;
