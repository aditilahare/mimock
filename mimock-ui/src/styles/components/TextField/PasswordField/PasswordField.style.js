import styled from 'styled-components';
import { UserNameFieldContainer } from "../UserNameField/UserNameField.style"

export const PasswordFieldContainer = styled(UserNameFieldContainer)
``;