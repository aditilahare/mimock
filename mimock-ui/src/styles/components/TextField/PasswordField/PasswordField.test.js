import React from 'react';
import { render } from '@testing-library/react';
import PasswordField from './PasswordField';

describe('Password', () => {
	let tree;

	it('should render password text field', async () => {
		tree = await render(
			<PasswordField
				dataTestid='login-password-input'
				placeholder='password'
				required
				autoComplete='off'
				value='password'
				onChangeHandler={() => {
					console.log('Change handler');
				}}
				onBlurHandler={() => {
					console.log('Blur handler');
				}}
				onFocusHandler={() => {
					console.log('Focus handler');
				}}
			/>
		);

		const { getByTestId, container } = tree;

		expect(getByTestId('login-password-input')).toBeInTheDocument();
		expect(container).toMatchSnapshot();
	});

	// it('should show password', async () => {
	// 	tree = await render(
	// 		<PasswordField
	// 			dataTestid='password-textfield'
	// 			errorLabel='Password is required'
	// 			isError={false}
	// 			value='password'
	// 			onChangeHandler={() => {
	// 				console.log('Change handler');
	// 			}}
	// 			onBlurHandler={() => {
	// 				console.log('Blur handler');
	// 			}}
	// 			onFocusHandler={() => {
	// 				console.log('Focus handler');
	// 			}}
	// 		/>
	// 	);

	// 	const { getByTestId, container } = tree;
	// 	const passwordInput = getByTestId('outlined-password-input');
	// 	const showPasswordButton = getByTestId('show-password-icon');
	// 	expect(passwordInput).toBeInTheDocument();
	// 	expect(showPasswordButton).toBeInTheDocument();

	// 	act(() => {
	// 		fireEvent.click(showPasswordButton);
	// 	});

	// 	expect(getByTestId('password-textfield')).toBeInTheDocument();
	// 	expect(passwordInput.children[0].type).toStrictEqual('text');
	// 	expect(container).toMatchSnapshot();
	// });
});
