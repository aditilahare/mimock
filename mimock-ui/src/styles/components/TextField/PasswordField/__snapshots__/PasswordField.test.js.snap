// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Password should render password text field 1`] = `
<div>
  <input
    autocomplete="off"
    class="w-full my-2 p-2 outline-none rounded-md border-2 border-gray-300 sc-bdvvtL"
    data-testid="login-password-input"
    placeholder="password"
    value="password"
  />
</div>
`;
